lib/puppet/application/doc.rb

```diff
@@ -40,7 +40,8 @@ def preinit
 
   option("--list", "-l") do |arg|
     require 'puppet/util/reference'
-    puts Puppet::Util::Reference.references.collect { |r| Puppet::Util::Reference.reference(r).doc }.join("\n")
+    refs = Puppet::Util::Reference.references(Puppet.lookup(:current_environment))
+    puts refs.collect { |r| Puppet::Util::Reference.reference(r).doc }.join("\n")
```

Comment:
prefer `map` over `collect`

Reply:
Since this is existing code, I'd rather not make that change. The maps are leading collects by about 2-1, but that seems like a bigger issue than this PR:

```
$ git grep "\.collect" lib | wc -l
     285
$ git grep "\.map" lib | wc -l
     466
```